sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala

```diff
@@ -1187,6 +1188,20 @@ class CachedTableSuite extends QueryTest with SQLTestUtils
     }
   }
 
+  test("cache supports for YearMonthIntervalType and DayTimeIntervalType") {
+    withTable("ymi_dti_interval_cache") {
+      Seq((1, Period.ofYears(1), Duration.ofDays(1)),
+        (2, Period.ofYears(2), Duration.ofDays(2)))
+        .toDF("k", "v1", "v2").write.saveAsTable("ymi_dti_interval_cache")
```

Comment:
for testing purpose, we can just do `...toDF(...).createTempView` and then cache the temp view.

Reply:
I thoroughly enjoyed this review.